File: Two_Sum/TwoSum.cs
public class Solution
{
    public int[] TwoSum(int[] nums, int target)
    {
        int[] result = new int[2];
        var numMap = new Dictionary<int, int>();

        for (int i = 0; i < nums.Length; i++)
        {
            int num = nums[i];
            if (numMap.ContainsKey(target - num))
            {
                result[0] = numMap[target - num];
                result[1] = i;
                return result;
            }
            numMap[num] = i;
        }
        return result;
    }
}

File: Two_Sum/TwoSum.py
from typing import List


class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        numMap = {}
        result = []
        for i in range(len(nums)):
            num = nums[i]
            if target - num in numMap:
                result.append(numMap[target - num])
                result.append(i)
                return result
            numMap[num] = i
        return result
